fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt

```diff
@@ -322,6 +322,11 @@ class Core(
                         rootStorageDirectory = context.filesDir,
                         readJson = readJson,
                         collectionName = COLLECTION_NAME,
+                        serverUrl = if (context.settings().serpTelemetryServer) {
+                            REMOTE_MAIN_ENDPOINT_URL
+                        } else {
+                            REMOTE_TEST_ENDPOINT_URL
+                        },
```

Comment:
If we turned this into a lambda, we could change it during runtime. We would also need to rebuild the `RemoteSettingsClient`, but I think that should be fine. For example

```
class SerpTelemetryRepository(
    rootStorageDirectory: File,
    private val readJson: () -> JSONObject,
    collectionName: String,
    serverUrl: { String } = { REMOTE_MAIN_ENDPOINT_URL, }
)

// here
serverUrl = { if (context.settings().serpTelemetryServer) {
        REMOTE_MAIN_ENDPOINT_URL
    } else {
        REMOTE_TEST_ENDPOINT_URL
    },
}